src/app/actions.ts

```diff
@@ -19,23 +17,43 @@ export type FlattenedRegion = {
   fullHierarchyName: string;
 };
 
-const fuseOptions = {
+export async function getRegion(regionCode: string) {
+  // We're reassigning the imported flattenedRegionCodes to a new variable
+  // here exclusively so that we can cast it to a Region[] type and get
+  // good type checking in the rest of the function.
+  const regions = flattenedRegionCodes as Region[];
+
+  const region = regions.find((region) => region.code === regionCode);
+
+  if (!region) {
+    throw new Error(`Could not find region with code ${regionCode}`);
+  }
+
+  return region;
+}
+
+export type FindRegionsResponse = {
+  item: Region;
+};
+
+const regionsFuseIndex = new Fuse<Region>(flattenedRegionCodes, {
   includeScore: true,
   keys: ["name"],
-};
-const regionsFuseIndex = new Fuse<FlattenedRegion>(
-  flattenedRegionCodes,
-  fuseOptions,
-);
+});
```

Comment:
I moved the fuse options for region search from their own variable to just passing the object inline. I don't think it adds much to readability to have them in their own variable and we only ever use them here so I don't think it's necessary for sharing them.

Reply:
Yeah that's neat, makes sense.